weave/api.py

```diff
@@ -327,22 +329,22 @@ def serve(
     trace_attrs = _monitor._attributes.get()
 
     def run():
-        with _wandb_api.wandb_api_context(wandb_api_ctx):
-            with attributes(trace_attrs):
-                uvicorn.run(app, host="0.0.0.0", port=port)
+        with _graph_client_context.set_graph_client(client):
+            with _wandb_api.wandb_api_context(wandb_api_ctx):
+                with attributes(trace_attrs):
+                    uvicorn.run(app, host="0.0.0.0", port=port)
 
     if _util.is_notebook():
         thread = True
     if thread:
-        from threading import Thread
 
-        t = Thread(target=run, daemon=True)
+        t = threading.Thread(target=run, daemon=True)
         t.start()
         time.sleep(1)
         return "http://localhost:%d" % port
     else:
         run()
-    return None
+    raise ValueError("Should not reach here")
```

Comment:
seems like you might want to put a return statement after the `run()`?

Reply:
run() should not return, uvicorn.run doesn't return